backfill/2022-01-27-sanitize-search-engines/README.md

```diff
@@ -0,0 +1,826 @@
+# Sanitize search engine values in historical data
+
+Investigation log for https://bugzilla.mozilla.org/show_bug.cgi?id=1751979
+
+It must match the pipeline sanitization logic in [MessageScrubber](https://github.com/mozilla/gcp-ingestion/blob/main/ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/MessageScrubber.java),
+particularly the `processForBug1751955` and `processForBug1751753` methods there.
+
+## Status
+
+As of 2022-02-16.
+
+`main_v4` has now been fully sanitized via Shredder.
+
+`main_summary_v4` is currently being sanitized via Shredder.
+
+All mobile stable tables have been sanitized; whd ran the following on 2022-02-18 (took ~5 minutes):
+
+```bash
+bq cp -f moz-fx-data-backfill-20:org_mozilla_fenix_stable.metrics_v1 moz-fx-data-shared-prod:org_mozilla_fenix_stable.metrics_v1
+bq cp -f moz-fx-data-backfill-20:org_mozilla_firefox_beta_stable.metrics_v1 moz-fx-data-shared-prod:org_mozilla_firefox_beta_stable.metrics_v1
+bq cp -f moz-fx-data-backfill-20:org_mozilla_fenix_nightly_stable.metrics_v1 moz-fx-data-shared-prod:org_mozilla_fenix_nightly_stable.metrics_v1
+bq cp -f moz-fx-data-backfill-20:org_mozilla_fennec_aurora_stable.metrics_v1 moz-fx-data-shared-prod:org_mozilla_fennec_aurora_stable.metrics_v1
+bq cp -f moz-fx-data-backfill-20:org_mozilla_focus_stable.metrics_v1 moz-fx-data-shared-prod:org_mozilla_focus_stable.metrics_v1
+bq cp -f moz-fx-data-backfill-20:org_mozilla_focus_beta_stable.metrics_v1 moz-fx-data-shared-prod:org_mozilla_focus_beta_stable.metrics_v1
+bq cp -f moz-fx-data-backfill-20:org_mozilla_focus_nightly_stable.metrics_v1 moz-fx-data-shared-prod:org_mozilla_focus_nightly_stable.metrics_v1
+bq cp -f moz-fx-data-backfill-20:org_mozilla_klar_stable.metrics_v1 moz-fx-data-shared-prod:org_mozilla_klar_stable.metrics_v1
+bq cp -f moz-fx-data-backfill-20:org_mozilla_firefox_stable.metrics_v1 moz-fx-data-shared-prod:org_mozilla_firefox_stable.metrics_v1
+```
+
+`clients_daily_v6` is backfilled and validated, ready for partitions 2019-11-22 through 2022-02-17
+to be copied into place.
```

Comment:
cc @whd

Reply:
done, via `dategen 20191122 20220217 | xargs -I{} -P 16 bq --project_id moz-fx-data-shared-prod cp -f 'moz-fx-data-backfill-20:telemetry_derived.clients_daily_v6${}' 'moz-fx-data-shared-prod:telemetry_derived.clients_daily_v6${}'`

where `dategen` is a script I have lying around that does the thing you might expect